### DISCUSSION TITLE
Ubuntu repository down?

### DISCUSSION THREAD

**Discussion by nShepard204** [i3/i3#5894] [Q&A] [answered]
Hello! I'm trying to update i3 to 4.22 on Ubuntu, and I've tried going through the documentation found here: https://i3wm.org/docs/repositories.html

This is what I'm getting in response
![image](https://github.com/i3/i3/assets/71411118/f8390809-2d9b-4726-ae93-e1988040856f)

I'm currently running this on an Ubuntu 22.04 LTS virtual machine. Not sure if I'm doing something wrong or if the repo's just having a fit. Has anyone else run into this problem?

Thanks in advance for the help!

**Comment by orestisfl** [comment]
@sur5r

**Comment by sur5r** [comment]
That's odd. I can't reproduce this in a jammy chroot. I also have no idea what that error message actually means.

I'm afraid I'm a bit lost here. Could you maybe strace/ltrace it?

```
strace -f -o apt-helper.tracelog /usr/lib/apt/apt-helper download-file https://debian.sur5r.net/i3/pool/main/s/sur5r-keyring/sur5r-keyring_2023.02.18_all.deb keyring.deb SHA256:a511ac5f10cd811f8a4ca44d665f2fa1add7a9f09bef238cdfad8461f5239cc4
```
or
```
ltrace -S -f -o apt-helper.tracelog /usr/lib/apt/apt-helper download-file https://debian.sur5r.net/i3/pool/main/s/sur5r-keyring/sur5r-keyring_2023.02.18_all.deb keyring.deb SHA256:a511ac5f10cd811f8a4ca44d665f2fa1add7a9f09bef238cdfad8461f5239cc4
```

and attach the resulting apt-helper.tracelog here.

**Reply by nShepard204** [reply]
Yeah sure thing! Should be attached to this reply.
[apt-helper.tracelog.txt](https://github.com/i3/i3/files/14128024/apt-helper.tracelog.txt)

Let me know if you need anything else from me

**Comment by sur5r** [comment]
The log contains a truncated error message. Can you rerun strace with `-s 1000` please?

**Reply by nShepard204** [reply]
Not a problem, here it is
[apt-helper.tracelog.txt](https://github.com/i3/i3/files/14142019/apt-helper.tracelog.txt)

**Comment by sur5r** [comment]
It seems this breaks somewhere during/after TLS handshake. Is the VM otherwise up to date?

Can you try downloading the keyring.deb using curl and/or wget? While that won't fix APT it might shed some light on whether TLS is working.

If that doesn't give any hints, could you please trace it using ltrace as well?

**Reply by nShepard204** [reply]
Yes, the VM is updated to the best of my knowledge

This is the verbose output from using curl
![image](https://github.com/i3/i3/assets/71411118/4104cd8f-ffec-492c-9a6a-ee0747f7e17e)

And the attached text file is the output from ltrace (using the command you provided earlier)
[apt-helper.tracelog.txt](https://github.com/i3/i3/files/14143078/apt-helper.tracelog.txt)

**Reply by sur5r** [reply]
This is getting increasingly weird. Is there anything/anyone that could be intercepting your traffic?

I guess this needs tcpdump level debugging. Feel free to contact me via eMail (sur5r@sur5r.net) if you don't want to share your public IP address here.

**Reply by sur5r** [reply]
I noticed something odd in that screenshot: As you copied everything from apt-download-helper to curl, curl tried to connect to a host called "keyring.deb". This doesn't exist on the public internet. ".deb" is not a valid TLD. Yet, on your machine, it resolves to a DigitalOcean IP. Something is broken regarding DNS as well.

**Comment by sur5r** [comment]
After inspecting tcpdump results there must be something intercepting the traffic. I can't help further.

**Comment by nShepard204** [comment, marked as answer]
Upon further inspection it seems like the problem here is my ISP. It appears to flag the repo site as a phishing site. Frankly if I end up caring that much about the gaps I'll just end up building it from source.

Thanks for the help!

**Reply by sur5r** [reply]
Holy. Fucking. Crap. Get yourself a proper ISP. This kind of traffic interception is a red flag. You shouldn't give them any money.
